### ISSUE TITLE
Stabilize Double Poisson training procedure

### ISSUE THREAD

**Issue by spencermyoung513** [porterjenkins/deep_uncertainty#13]
In our brief experiments training neural networks with the Double Poisson NLL, we have observed a large variance in the resultant model's metrics (primarily MSE). This could be due to a variety of factors. The goal is to design a training process with the Double Poisson NLL that consistently outputs good models (with low variance in terms of model MSE across random initializations).

Areas to tinker with:

- [ ] Model initialization (different random initializations for model weights, along with schematic adjustments to the bias of the final layer to account for empirical statistical parameters)
- [ ] Model architecture (residual connections, batch normalization, etc.)
- [x] Optimization algorithm (Adam vs. AdamW vs. SGD, learning rate warmup / cosine annealing / other scheduling, etc.)
- [x] Training time (number of epochs)

**Comment by spencermyoung513** [comment]
Got some encouraging results with the following training settings:

```
epochs: 2000
batch_size: 32
lr: 1e-3
weight_decay: 1e-5
optimizer: AdamW
lr_scheduler: CosineAnnealingLR
```
I also trained each network 10 times. Crossing my fingers that we've found some stability.

![mean_calibration_dists](https://github.com/porterjenkins/deep_uncertainty/assets/76703541/48f80dd2-bd0f-48ec-a34f-6f79818c9933)
![med_se_dists](https://github.com/porterjenkins/deep_uncertainty/assets/76703541/7cdeeaf4-0526-4e72-bbec-3828f1cad47c)
![mse_dists](https://github.com/porterjenkins/deep_uncertainty/assets/76703541/1542a31a-f347-4335-8e79-22eb0063f936)
